iceoryx_hoofs/source/cxx/requires.cpp

```diff
@@ -19,15 +19,17 @@
 #include "iceoryx_hoofs/error_handling/error_handling.hpp"
 
 #include <iostream>
+#include <cstdint>
 
 namespace iox
 {
 namespace cxx
 {
 namespace internal
 {
+// AXIVION Next Line  AutosarC++19_03-A3.3.1 : The function is declared with in the namespace
 void Require(
-    const bool condition, const char* file, const int line, const char* function, const char* conditionString) noexcept
+    const bool condition, const int8_t* file, const int32_t line, const int8_t* function, const int8_t* conditionString) noexcept
```

Comment:
I would not change the `char*` to `int8_t`. It is just confusing. I am not sure `char` is always one byte, though we do not need this guarantee here anyway as far as I can see (it only needs to be consistent).

`const int32_t line` is ok though. However it is maybe always called with \_\_LINE\_\_ which provides an `int` (I think it is implementation/compiler dependent) so efficiency may suffer.

Reply:
`char*` is not an `int8_t*`. An `int8_t*` is a `signed char` and this comes with problems since a string literal is always deduced as a `const char*`. A `char` is even a static size type and not like `int` where the size varies on different platforms, see: https://isocpp.org/wiki/faq/intrinsic-types#:~:text=Even%20if%20you%20think%20of,No%20exceptions,%20ever.